lib/matplotlib/backend_bases.py

```diff
@@ -3642,6 +3651,36 @@ def trigger_tool(self, name):
         """
         self.navigation.tool_trigger_event(name, sender=self)
 
+    def get_group_start_position(self, name):
+        count = 0
+        for group in self.groups:
+            if group == name:
+                break
+            count += len(self.groups[group])
+        return count
+
+    def add_tool_to_group(self, name, group, position=-1):
+        """Adds the tool to the group
+
+        Parameters
+        ----------
+        name : string
+            The name of the tool.
+        group : string
+            The name of the group.
+        position : int
+            The position within the group, if -1, it goes at the end.
+        """
```

Comment:
Note to self, need to add some more documentation here, i.e. say it returns the position that this should get inserted into the toolbar.